File: hook_system/client/ClientServer/Models/Course.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace ClientServer.Models
{
    public class Course
    {
        public long CourseId { get; set; }
        public string UserId { get; set; }
        public int Year { get; set; }
        public int Semester { get; set; }
        public string ProgramCode { get; set; }
        public string CourseCode { get; set; }
        
        // [JsonIgnore]
        public virtual List<Assignment> Assignments { get; set; }
    }
}

File: hook_system/client/ClientServer/ClientApp/src/app/shared/models/course.ts
export class Assignment {
    assignmentId: number;
    name: string;
    courseId: number;
    openDate: Date;
    closeDate: Date;
    status: boolean;
    course: Course;
}

export class Course {
    courseId: number;
    userId: number;
    year: number;
    semester: number;
    programCode: string;
    courseCode: string;

    assignments: Assignment[];
}